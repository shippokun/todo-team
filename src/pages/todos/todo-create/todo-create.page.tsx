import { TodoCreateContainer } from "./containers";

export const TodoCreatePage: React.FC = () => {
  return <TodoCreateContainer />;
};
