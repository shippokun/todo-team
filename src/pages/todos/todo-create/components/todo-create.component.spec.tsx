import * as React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";

import { TodoCreate } from "./todo-create.component";

describe("todo-create.component.tsx", () => {
  const setup = () => {
    const utils = render(<TodoCreate />);
    const inputTitle = utils.getByLabelText("title-input") as HTMLInputElement;
    const submit = utils.getByDisplayValue("送信") as HTMLInputElement;
    return {
      inputTitle,
      submit,
      utils,
    };
  };

  afterEach(() => {
    cleanup();
  });

  it("render", () => {
    const { asFragment } = render(<TodoCreate />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("input text", () => {
    const { inputTitle } = setup();
    const inputText = "test";
    fireEvent.change(inputTitle, { target: { value: inputText } });
    expect(inputTitle.value).toBe(inputText);
  });

  it.skip("submit event", () => {
    // TODO: submit eventの発火したかどうかのテスト
  });
});
