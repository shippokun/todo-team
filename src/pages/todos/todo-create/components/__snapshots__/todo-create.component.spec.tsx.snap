// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo-create.component.tsx render 1`] = `
<DocumentFragment>
  <form
    style="text-align: start;"
  >
    <p>
      タイトル：
      <input
        aria-label="title-input"
        name="title"
      />
    </p>
    <p>
      <input
        type="submit"
        value="送信"
      />
    </p>
  </form>
</DocumentFragment>
`;
